services/venice-server/src/main/java/com/linkedin/venice/listener/OutboundHttpWrapperHandler.java

```diff
@@ -55,18 +55,16 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
     try {
       if (msg instanceof AbstractReadResponse) {
         AbstractReadResponse obj = (AbstractReadResponse) msg;
-        ServerStatsContext statsContext = statsHandler.getServerStatsContext();
-        setStats(statsContext, obj);
-
+        requestStatsRecorder.setReadResponseStats(obj.getStatsRecorder());
```

Comment:
While I'm not opposed to renaming the `ServerStatsContext`, I think we may need to take a holistic look at the various "stats recorders" we have in play. There is a "read response stats recorder" since my last refactoring, and now we would have a "request stats recorder" as well. Maybe this is fine, or maybe it needs some further tweaking, I am not sure. But this line in particular illustrates the confusion (it seems that at the very least, the `AbstractReadResponse::getStatsRecorder` function ought to be renamed more precisely?)

Reply:
IMHO we should merge `RequestStatsRecorder` (formerly `ServerStatsContext`)  and all of `ReadResponseStatsRecorder` into a common class to keep things simple. 

However, if we think it's not desriable I can change `AbstractReadResponse::getStatsRecorder` to `AbstractReadResponse::getReadResponseStatsRecorder` or something?

Reply:
I do have one question: why did we introduce `ReadResponseStatsRecorder`?

Reply:
So... my motivation during that refactoring was that I did not like the previous structure of the code where a bunch of stats were getting recorded in the `ReadResponse`, and then had to be copied one by one into the `ServerStatsContext`, and then had to be taken from that context and passed into the various metric recording calls... I thought it was kind of tedious to add one more metric, having to wire through all these steps.

In the new code structure, stats are accumulated in a `ReadResponseStats`, which only carries setters and incrementer methods. And then a reference to that one object is passed into the context one time (not each field separately). Finally, when it's recording time, the `ReadResponseStatsRecorder` comes into play and accepts the metrics recorder object and acts on it.

I think it could be fine to pull more of the `ServerStatsContext` fields into the `ReadResponseStats`... or maybe there are more/bigger structural improvements we could do. IDK.

Reply:
I've rename `AbstractReadResponse::getStatsRecorder` to `AbstractReadResponse::getReadResponseStatsRecorder` as suggested in the first comment. 

As for larger/broader refactoring of stats record, I would like to revisit it in the future